spack/packages/dla-future/package.py

```diff
@@ -109,7 +109,12 @@ def cmake_args(self):
 
         # BLAS/LAPACK
         if "^mkl" in spec:
-            args.append(self.define("DLAF_WITH_MKL", True))
+            vmap = {"none": "seq", "openmp": "omp", "tbb": "tbb"} # Map MKL variants to LAPACK target name
+            # TODO: Generalise for intel-oneapi-mkl
+            args += [
+                self.define("DLAF_WITH_MKL", True),
+                self.define("MKL_LAPACK_TARGET", "mkl::mkl_intel_32bit_{0}_dyn".format(vmap[spec["intel-mkl"].variants["threads"].value])),
+            ]
```

Comment:
This can be left as an excercise for another PR, but I just want to note this here.

We should most likely be using something like this instead: https://spack-tutorial.readthedocs.io/en/sc20/tutorial_advanced_packaging.html#accessing-dependency-libraries. This would remove the need for `MKL_LAPACK_TARGET`. We do still need some of saying "we're using MKL" to be able to call `mkl_set_num_threads`. This could be automatically detected, but simpler is probably just to set `DLAF_WITH_MKL` as now since we know in the spack package if `intel-mkl` or `intel-oneapi-mkl` is being used.

Reply:
@msimberg should I address this here, or do you want to raise a separate issue?

Reply:
I'd be happy with a separate issue.

Reply:
https://github.com/eth-cscs/DLA-Future/issues/851